index.js

```diff
@@ -90,6 +98,59 @@ function inject (dispatchFunc, options, callback) {
   }
 }
 
+function Chain (dispatch, option) {
+  this.option = Object.assign({}, option)
+  this.dispatch = dispatch
+  this._promise = null
+}
+
+const httpMethods = [
+  'delete',
+  'get',
+  'head',
+  'options',
+  'patch',
+  'post',
+  'put',
+  'trace'
+]
+
+httpMethods.forEach(method => {
+  Chain.prototype[method] = function (url) {
+    this.option.url = url
+    this.option.method = method.toUpperCase()
+    return this
+  }
+})
+
+const chainMethods = [
+  'body',
+  'headers',
+  'payload',
+  'query'
+]
+
+chainMethods.forEach(method => {
+  Chain.prototype[method] = function (value) {
+    this.option[method] = value
+    return this
+  }
+})
+
+Chain.prototype.end = function (callback) {
+  doInject(this.dispatch, this.option, callback)
+}
+
+Object.getOwnPropertyNames(Promise.prototype).forEach(method => {
+  if (method === 'constructor') return
+  Chain.prototype[method] = function (...args) {
+    if (!this._promise) {
+      this._promise = doInject(this.dispatch, this.option)[method](...args)
+    }
+    return this._promise
```

Comment:
you need to call the method on promise, passing the args to it anyway.

Reply:
Sorry for forgetting changing the code.